apple/internal/templates/apple_device.template.py

```diff
@@ -300,12 +300,22 @@ def extracted_app(
     # fail with `Unhandled error domain NSPOSIXErrorDomain, code 13`.
     dst_dir = os.path.join(tempfile.gettempdir(), "bazel_temp_" + app_name)
     os.makedirs(dst_dir, exist_ok=True)
+    file_mode = "u+w"
+    # Mitigate any previous runs of openrsync that synced the application
+    # into dst_dir without applying the correct permissions.
+    dst_dir_application = os.path.join(
+      dst_dir, os.path.basename(application_output_path)
+    )
+    if (os.path.exists(dst_dir_application)
+        and not os.access(dst_dir_application, os.W_OK)):
+        fix_permissions(dst_dir, file_mode)
     rsync_command = [
         "/usr/bin/rsync",
         "--archive",
         "--delete",
         "--checksum",
-        "--chmod=u+w",
+        "--perms",
```

Comment:
Adding `--perms` since the man page says it is required when passing `--chmod`.
Does not have any observable effect at the moment.